source/expectations/matchers/ut_be_within.tpb

```diff
@@ -0,0 +1,113 @@
+    l_result boolean;
+  begin
+    if self.expected.data_type = a_actual.data_type then
+      if self.expected is of (ut_data_value_number) and self.is_pct = 0 then
+        l_result := abs((treat(self.expected as ut_data_value_number).data_value - treat(a_actual as ut_data_value_number).data_value)) <= 
+                    treat(self.dist as ut_data_value_number).data_value;
+      elsif self.expected is of (ut_data_value_number) and self.is_pct = 1 then
+        l_result := treat(self.dist as ut_data_value_number).data_value >= 
+                    (
+                     ((treat(self.expected as ut_data_value_number).data_value - treat(a_actual as ut_data_value_number).data_value ) * 100 ) /
+                    (treat(self.expected as ut_data_value_number).data_value)) ;      
+      elsif self.expected is of (ut_data_value_date) and self.dist is of ( ut_data_value_yminterval) then      
+        l_result := treat(a_actual as ut_data_value_date).data_value 
+                    between 
+                     (treat(self.expected as ut_data_value_date).data_value) - treat(self.dist as ut_data_value_yminterval).data_value
+                     and 
+                     (treat(self.expected as ut_data_value_date).data_value) + treat(self.dist as ut_data_value_yminterval).data_value;
+      elsif self.expected is of (ut_data_value_date) and self.dist is of ( ut_data_value_dsinterval) then      
+        l_result := treat(a_actual as ut_data_value_date).data_value 
+                    between 
+                      (treat(self.expected as ut_data_value_date).data_value) - treat(self.dist as ut_data_value_dsinterval).data_value
+                      and 
+                      (treat(self.expected as ut_data_value_date).data_value) + treat(self.dist as ut_data_value_dsinterval).data_value;
+      end if;
+    else
+      l_result := (self as ut_matcher).run_matcher(a_actual);
+    end if;
+    return l_result;
+  end;
+
+  overriding member function failure_message(a_actual ut_data_value) return varchar2 is
+    l_distance varchar2(32767);
+  begin
+    l_distance := case 
+                    when self.dist is of (ut_data_value_number) then
+                      treat(self.dist as ut_data_value_number).to_string
+                    when self.dist is of (ut_data_value_yminterval) then
+                      treat(self.dist as ut_data_value_yminterval).to_string 
+                    when self.dist is of (ut_data_value_dsinterval) then
+                      treat(self.dist as ut_data_value_dsinterval).to_string 
+                    else
+                      null
+                    end;
+                    
+    return (self as ut_matcher).failure_message(a_actual) || ' '||l_distance ||' of '|| expected.to_string_report();
+  end;
+
+  overriding member function failure_message_when_negated(a_actual ut_data_value) return varchar2 is
```

Comment:
Shouldn't we calculate the distance here too? 
It's not covered by tests so hard to say